Polyfills/Canvas/Source/Image.cpp

```diff
@@ -97,13 +98,56 @@ namespace Babylon::Polyfills::Internal
         }
     }
 
+    bool NativeCanvasImage::SetBuffer(gsl::span<const std::byte> buffer)
+    {
+        m_imageContainer = bimg::imageParse(&Graphics::DeviceContext::GetDefaultAllocator(), buffer.data(), static_cast<uint32_t>(buffer.size_bytes()), bimg::TextureFormat::RGBA8);
+
+        if (m_imageContainer == nullptr)
+        {
+            return false;
+        }
+
+        m_width = m_imageContainer->m_width;
+        m_height = m_imageContainer->m_height;
+
+        if (!m_onloadHandlerRef.IsEmpty())
+        {
+            m_onloadHandlerRef.Call({});
+        }
+        return true;
+    }
+
     void NativeCanvasImage::SetSrc(const Napi::CallbackInfo& info, const Napi::Value& value)
     {
         auto text{value.As<Napi::String>().Utf8Value()};
+
+        // try with base64
+        static const std::string base64{"base64,"};
+        const auto pos = text.find(base64);
+        if (pos != std::string::npos)
+        {
+            arcana::make_task(m_runtimeScheduler, *m_cancellationSource, [env{info.Env()}, this, text{std::move(text)}, pos]() {
+                std::vector<uint8_t> base64Buffer;
+                bn::decode_b64(text.begin() + pos + base64.length(), text.end(), std::back_inserter(base64Buffer));
+                gsl::span<const std::byte> buffer = {reinterpret_cast<std::byte*>(base64Buffer.data()), base64Buffer.size()};
+
+                if (!SetBuffer(buffer))
+                {
+                    HandleLoadImageError(Napi::Error::New(env, "Unable to decode image with provided base64 source."));
+                }
+            });
+            return;
+        }
+
+        // try with URL
         UrlLib::UrlRequest request{};
         request.Open(UrlLib::UrlMethod::Get, text);
         request.ResponseType(UrlLib::UrlResponseType::Buffer);
-        request.SendAsync().then(m_runtimeScheduler, *m_cancellationSource, [env{info.Env()}, this, request{std::move(request)}](arcana::expected<void, std::exception_ptr> result) {
+        request.SendAsync().then(m_runtimeScheduler, *m_cancellationSource, [env{info.Env()}, this, cancellationSource{m_cancellationSource}, request{std::move(request)}, text](arcana::expected<void, std::exception_ptr> result) {
+            if (cancellationSource->cancelled())
+            {
+                return;
+            }
```

Comment:
This shouldn't be necessary. arcana.cpp will check this before calling the continuation.

Reply:
Fixed